### PR TITLE
Testing/Sample - Made checking confusion matrix data more robust

### PR DESCRIPTION
**Pull request by Ark-kun**
The sample tests no longer depend on particular  file names inside the archive. Now they only depend on the artifact name.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1196)

### CODE REVIEW

**Comment by gaoning777** [comment]
Could you debug the sample test error, please?

**Comment by Ark-kun** [comment]
> Could you debug the sample test error, please?

Thanks for noticing this.

Looks like `json.load` only supports reading from binary files since python 3.6. https://docs.python.org/3/library/json.html#json.load

I'll fix that.

**Comment by Ark-kun** [comment]
> Could you debug the sample test error, please?

Sample test succeeds now.

**Comment by gaoning777** [comment]
The link(https://docs.python.org/3/library/json.html#json.load) says it supports both text and binary.

File: test/sample-test/run_xgboost_test.py [hunk 1]
```diff
@@ -109,14 +109,14 @@ def main():
...
-  with open(cm_filename, 'r') as f:
-    cm_data = f.read()
-    utils.add_junit_test(test_cases, 'confusion matrix format', (len(cm_data) > 0), 'the confusion matrix file is empty')
+  utils.get_artifact_in_minio(workflow_json, 'confusion-matrix', cm_tar_path, 'mlpipeline-ui-metadata')
+  with tarfile.open(cm_tar_path) as tar_handle:
+    file_handles = tar_handle.getmembers()
+    assert len(file_handles) == 1
+
+    with tar_handle.extractfile(file_handles[0]) as f:
+      cm_data = f.read()
```
Comment (gaoning777) [inline test/sample-test/run_xgboost_test.py]: Why not use the io.TextIOWrapper here as well?

Comment (Ark-kun) [reply test/sample-test/run_xgboost_test.py]: I'm not sure it's needed.
I can add it, but it won't change anything. It's mostly needed for older `json.load` which cannot read from files opened in binary mode.

**Comment by Ark-kun** [comment]
> The link(https://docs.python.org/3/library/json.html#json.load) says it supports both text and binary.

That's what I was thinking. Unfortunately, the footnote at that link is:

>Changed in version 3.6: fp can now be a binary file. The input encoding should be UTF-8, UTF-16 or UTF-32.

**Pull request merged**